test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -283,4 +286,140 @@ public void multipleSubscriberCBRTest() throws Exception {
+
+        // Write 7 events to the stream.
+        for (int i = 1; i <= 7; i++) {
+            log.info("Writing event to {}/{}", SCOPE_1, STREAM_1);
+            writer.writeEvent(SIZE_30_EVENT).join();
+        }
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE_1, clientConfig);
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig
+                .builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints().stream(Stream.of(SCOPE_1, STREAM_1)).build();
+
+        assertTrue("Reader group is not created", readerGroupManager.createReaderGroup(READER_GROUP_3, readerGroupConfig));
+        assertEquals(1, controller.listSubscribers(SCOPE_1, STREAM_1).join().size());
+
+        @Cleanup
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(READER_GROUP_3);
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(READER_GROUP_3 + "-" + 1,
+                READER_GROUP_3, new JavaSerializer<>(), readerConfig);
+
+        EventRead<String> read;
+        // Read three events with reader.
+        for (int i = 1; i <= 3; i++) {
+            read = reader.readNextEvent(READ_TIMEOUT);
+            assertEquals("data of size 30", read.getEvent());
+        }
+        log.info("{} generating 1st stream-cuts for {}/{}", READER_GROUP_3, SCOPE_1, STREAM_1);
+        CompletableFuture<Map<Stream, StreamCut>> futureCuts = readerGroup.generateStreamCuts(streamCutExecutor);
+        // Wait for 5 seconds to force reader group state update. This will allow for the silent
+        // checkpoint event generated as part of generateStreamCuts to be picked and processed.
+        Futures.delayedFuture(Duration.ofSeconds(5), executor).join();
+        read = reader.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read.getEvent());
+        assertTrue("Stream-cut generation did not complete for reader group", Futures.await(futureCuts, 10000));
+        Map<Stream, StreamCut> streamCuts = futureCuts.join();
+
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP_3, streamCuts);
+        readerGroup.updateRetentionStreamCut(streamCuts);
+
+        // Retention set has one stream cut at 0/210
+        // READER_GROUP_3 updated stream cut at 0/90
+        // Subscriber lower bound is 0/90, truncation should happen at this point
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 90.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(SCOPE_1, STREAM_1), 0L).join().values().stream().anyMatch(off -> off == 90),
+                5000, 3 * 60 * 1000L);
```

Comment:
Do you know how long takes truncation to happen here? If it takes more than 30 seconds, we may need to see if the configuration of needs to be changed to allow truncations to happen more frequently. Otherwise, the test is going to take too long for such a small workload.

Reply:
There is a property `controller.retention.frequency.minutes=2` which says retention is happening every 2 minutes. Not sure on a way to make it in seconds.

Reply:
Reduced the retention frequency to 1 minute.